internal/pkg/name-conversions/conversions.go

```diff
@@ -0,0 +1,90 @@
+package nameconversions
+
+import (
+	"fmt"
+
+	"github.com/confluentinc/cli/internal/pkg/errors"
+)
+
+const (
+	ResourceNameNotFoundErrorMsg          = `resource with name or ID "%s" was not found`
+	DuplicateResourceNameErrorMsg         = `the resource name "%s" is shared across multiple resources`
+	DuplicateResourceNameErrorSuggestions = "retry the previous command using a resource id"
+)
+
+type resource any
+
+type resourcePtr interface {
+	GetId() string
+}
+
+type specPtr interface {
+	GetDisplayName() string
+}
+
+type v2ResourcePtr interface {
+	GetDisplayName() string
+	resourcePtr
+}
+
+func ConvertToPtrSlice[V resource](resources []V) []*V {
+	ptrs := make([]*V, len(resources))
+	for i := range resources {
+		ptrs[i] = &resources[i]
+	}
+	return ptrs
+}
+
+// ConvertSpecNameToId ConvertNamesToID returns a resource spec's name's corresponding ID or returns an error if not found
+func ConvertSpecNameToId[V resourcePtr, T specPtr](input string, resources []V, specs []T) (string, error) {
+	namesToIds, err := GetSpecNamesToIds(resources, specs)
+	if err != nil {
+		return input, err
+	}
+	if resourceId, ok := namesToIds[input]; ok {
+		return resourceId, nil
+	} else {
+		return input, errors.Errorf(ResourceNameNotFoundErrorMsg, input)
+	}
+}
+
+// GetSpecNamesToIds returns a mapping from spec resource names to their respective IDs
+func GetSpecNamesToIds[V resourcePtr, T specPtr](resources []V, specs []T) (map[string]string, error) {
+	namesToIds := make(map[string]string, len(resources))
+	for i := range resources {
+		name := specs[i].GetDisplayName()
+		if _, ok := namesToIds[name]; !ok {
+			namesToIds[name] = resources[i].GetId()
+		} else {
+			return nil, errors.NewErrorWithSuggestions(fmt.Sprintf(DuplicateResourceNameErrorMsg, name), DuplicateResourceNameErrorSuggestions)
+		}
```

Comment:
This will return an error if any duplicated name is found, even if the user is attempting to work with a non-duplicated name.

While not an issue for resources like service accounts or environments where the backend enforces unique names, duplicate names can happen with Kafka clusters.

One way to resolve this is by using a `map[string][]string` where the slice value stores all id's with that name, and then you can throw the duplicate error if the length of the slice that the user is accessing is `> 1`. There's probably more elegant solutions, though.

Reply:
Yeah, I didn't think about that case. We definitely don't want to give the user an error unless it is the resource name they are currently using that is shared by two different resource IDs